packages/puppeteer-core/src/cdp/NetworkManager.test.ts

```diff
@@ -16,10 +18,32 @@ import {NetworkManagerEvent} from '../common/NetworkManagerEvents.js';
 
 import type {CdpFrame} from './Frame.js';
 import {NetworkManager} from './NetworkManager.js';
-
 // TODO: develop a helper to generate fake network events for attributes that
 // are not relevant for the network manager to make tests shorter.
 
+interface CdpLog extends Record<string, unknown> {
+  method: string;
+  params: any;
+}
+
+async function runCdpLogSnapshot(
+  cdpSession: MockCDPSession,
+  name: `${string}.json`
+) {
+  const json = await readFile(
+    resolve(__dirname, '../../../../', './src/cdp/snapshots', name),
+    'utf-8'
+  );
+  const events = JSON.parse(json) as CdpLog[];
+
+  for (const event of events) {
+    cdpSession.emit(event.method, event.params);
```

Comment:
```suggestion
    setImmediate(() => {
       cdpSession.emit(event.method, event.params);
    })
```

And remove the setTimeout